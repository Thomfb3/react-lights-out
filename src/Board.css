.Board {
  margin:0 auto;
  border-spacing: 10px;
  position: relative;
  top: 150px;
}

.Board-title {
  position: relative;
  top: 40px;
  margin: 0px;
  font-family: 'Righteous', cursive;
  font-size: 60px;
  color:#94a9fe;
  text-shadow: 0px 0px 15px rgba(205, 217, 255, 0.2);
}