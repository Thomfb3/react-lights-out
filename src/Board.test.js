import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Board from "./Board";


it("renders without crashing", function () {
    render(<Board />);
});

it("should match snapshot", function () {
    const { asFragment } = render(<Board />);
    expect(asFragment()).toMatchSnapshot();
});


it("should render the board's table ", function () {
    const { getByTestId } = render(<Board />);
    expect(getByTestId("Board")).toBeInTheDocument();
});


it("should not render the game won ", function () {
    const { getByTestId } = render(<Board />);
    expect(getByTestId("Board")).not.toContainHTML('<h1 className="GameWon">You won!</h1>');
});
