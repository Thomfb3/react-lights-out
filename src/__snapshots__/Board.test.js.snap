// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <table
    data-testid="Board"
  >
    <tbody
      class="Board"
    >
      <tr>
        <td
          class="Cell "
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell "
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell "
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell "
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
